internal/nvmeof/nvmeof.go

```diff
@@ -44,36 +43,37 @@ type ListenerDetails struct {
 	Hostname string
 }
 
+// Config holds gateway client configuration.
+type GatewayConfig = GatewayAddress
+
 // NVMeoFVolumeData holds the data required for an NVMe-oF volume.
 type NVMeoFVolumeData struct {
 	SubsystemNQN          string
 	NamespaceID           uint32
 	HostNQN               string
 	ListenerInfo          ListenerDetails
-	GatewayManagementInfo GatewayAddress
-}
-
-// Config holds gateway client configuration.
-type GatewayConfig struct {
-	GatewayAddress
+	GatewayManagementInfo GatewayConfig
 }
 
 // GatewayClient wraps the gRPC client and connection details.
 type GatewayRpcClient struct {
-	config *GatewayConfig
-	conn   *grpc.ClientConn
-	client GatewayClient // protobuf client
-
-	// Thread-safe serial generation
-	serialMutex sync.Mutex
-	maxSerial   *big.Int
+	config    *GatewayConfig
+	conn      *grpc.ClientConn
+	client    GatewayClient // protobuf client
+	maxSerial *big.Int      // Maximum allowed serial number for subsystems.
 }
 
+// TODO: Move this to a util package?
+const (
+	MaxAllowedSubsystemSerialNumber = 99999999999999
+	SerialOffset                    = 2 // prevent 0 and 1 as serial numbers
```

Comment:
Should this offset be stored somewhere? What happens if the controller restarts, and the offset is reset back to it's initial value?